src/autopas/containers/cellPairTraversals/SlicedBlkBasedTraversal.h

```diff
@@ -0,0 +1,953 @@
+  std::map<std::array<unsigned long, 3>, unsigned long> _cellBlocksToIndex;
+  /**
+   * A map converting a cellblock index to their relative cellblock coordinates.
+   */
+  std::map<unsigned long, std::array<unsigned long, 3>> _indexToCellBlock;
+
+  std::string debugHelperFunctionIntArrayToString(std::array<std::vector<unsigned long>, 3> _cellBlockDimensions) {
+    std::string str;
+    for (int j = 0; j < 3; ++j) {
+      str += std::to_string(j) + ": ";
+      for (auto i : _cellBlockDimensions[j]) {
+        str += std::to_string(i) + "| ";
+      }
+    }
+    return str;
+  }
+  std::string debugHelperFunctionIntArrayToString(std::array<unsigned long, 3> intArray) {
+    std::string str;
+    for (auto i : intArray) {
+      str += std::to_string(i) + " ";
+    }
+    return str;
+  }
+  std::string debugHelperFunctionIntArrayToString2(std::array<unsigned long, 3> intArray) {
+    std::string str;
+    for (auto i : intArray) {
+      str += std::to_string(i) + " ";
+    }
+    return str;
+  }
+  std::string debugHelperFunctionIntArrayToString(std::array<double, 3> intArray) {
+    std::string str;
+    for (auto i : intArray) {
+      str += std::to_string(i) + " ";
+    }
+    return str;
+  }
+
+  /**
+   * Testing the locking of all neighbour sub-blocks of the given one, including the given one.
+   *
+   * @param blockNumber The number of the block/ thread which wants the sub-block to lock.
+   * @param order The block unique order id of the sub-block. e.g. 000-222 as an array. = {0,0,0}-{2,2,2}
+   * @return A boolean corresponding true if the locks could been set, or false otherwise.
+   */
+  bool locking2x2x2sub_blocks(unsigned long blockNumber, std::array<unsigned long, 3> order) {
+    // IF LOCK POSSIBLE, LOCK & SET MASTER LOCK & return true
+    // IF IMPOSSIBLE UNLOCK locked Locks & SET MASTER LOCK & return false
+    bool LocksThatCanBeSet[8] = {false, false, false, false, false, false, false, false};
+    int no_locksSet = 0;
+    _masterlock.lock();
+    //    AutoPasLog(debug, "LOCKING: " + std::to_string(blockNumber) + " MASTER");
+
+    for (unsigned long x = 0; x < 2; ++x) {
+      for (unsigned long y = 0; y < 2; ++y) {
+        for (unsigned long z = 0; z < 2; ++z) {
+          LocksThatCanBeSet[no_locksSet] =
+              locks[uniqueLockCalculation(blockNumber, {x + order[0], y + order[1], z + order[2]})].testlock();
+          //          AutoPasLog(
+          //              debug,
+          //              "TRYING LOCKING " + std::to_string(blockNumber) + " Lock : " +
+          //                  std::to_string(uniqueLockCalculation(blockNumber, {x + order[0], y + order[1], z +
+          //                  order[2]})) + " id: " + debugHelperFunctionIntArrayToString2({x + order[0], y + order[1],
+          //                  z + order[2]}));
+          no_locksSet++;
+        }
+      }
+    }
+
+    if (LocksThatCanBeSet[0] && LocksThatCanBeSet[1] && LocksThatCanBeSet[2] && LocksThatCanBeSet[3] &&
```

Comment:
to be consistent with the rest of autopas replace all `&&` with `and` and all `||` with `or`